<launch>

  <arg name="room" default="dsa" />
  <arg name="world" default="floor" />
  <arg name="robot_name" default="r1" />
  <arg name="bridge_pose" default="true" />
  <arg name="bridge_command" default="true" />
  <arg name="bridge_segments" default="true" />
  <arg name="bridge_waypoints" default="true" />
  <arg name="bridge_gazebo" default="true" />
  <arg name="pose_gt" default="pose_gt" />  <!-- pose form gazebo ground truth -->
  <arg name="pose_tf" default="pose_tf" />  <!-- pose published via a TF (self-localization) -->

  
   <!-- Gazebo  -->
   <include file="$(find dsa_gazebo)/launch/all.launch" >
        <arg name="world" value="$(arg world)" />
        <arg name="room" value="$(arg room)" />
   </include>
  
   <!-- Laser bridge  -->
   <include file="$(find shmfw_laser_scan)/launch/laser_scan.launch" >
        <arg name="robot_name" value="$(arg robot_name)" />
   </include>
   
  <group ns="$(arg robot_name)">
    <node pkg="shmfw_bridge" type="shmfw_bridge_node" name="shmfw_bridge" output="screen">
      <param name="frequency" value="10"/> 
      <param name="pose/tf_prefix" value="$(arg robot_name)"/> 
      <param name="pose/target_frame" value="/map"/>    
      <param name="pose/source_frame" value="base_link"/>  
      <param name="pose/shm_name_pose" value="$(arg pose_tf)"/>  
      <param name="pose/shm_name_agent_state" value="agent_state"/>  
      <param name="cmd/shm_name_cmd" value="cmd_vel"/> 
      <param name="cmd/timeout_signal" value="0.5"/>     
      <param name="cmd/frequency" value="-1"/>     
      <param name="path/resolution" value="0.2"/>       
      <param name="cmd/frequency" value="-1"/>     
      <param name="gazebo/frequency" value="10"/>    
      <param name="gazebo/shm_name_pose" value="$(arg pose_gt)"/>    
      <param name="gazebo/shm_name_state" value="state_gt"/>    
      <param name="bridge_pose" value="$(arg bridge_pose)"/>    
      <param name="bridge_command" value="$(arg bridge_command)"/>    
      <param name="bridge_segments" value="$(arg bridge_segments)"/>    
      <param name="bridge_waypoints" value="$(arg bridge_waypoints)"/>    
      <param name="bridge_gazebo" value="$(arg bridge_gazebo)"/>    
    </node>

    <node pkg="shmfw_visualization_marker" type="shmfw_visualization_marker_node" name="shmfw_visualization_marker" output="screen">
      <param name="frequency" value="-1"/> 
      <param name="shm_marker_name_array" value="VisualizationMarkerArray"/>       
    </node>
  </group>

</launch>
